core/trino-main/src/main/java/io/trino/sql/planner/planprinter/TextRenderer.java

```diff
@@ -118,20 +118,23 @@ private String writeTextOutput(StringBuilder output, PlanRepresentation plan, In
     private String printStats(PlanRepresentation plan, NodeRepresentation node)
     {
         StringBuilder output = new StringBuilder();
-        if (node.getStats().isEmpty() || !(plan.getTotalCpuTime().isPresent() && plan.getTotalScheduledTime().isPresent())) {
+        if (node.getStats().isEmpty() || !(plan.getTotalCpuTime().isPresent() && plan.getTotalScheduledTime().isPresent() && plan.getTotalBlockedTime().isPresent())) {
```

Comment:
is that intentional to not print anything if any of the stats is missing?

Reply:
Stats are available when

1. Generating a distributed plan for `QueryCompletedEvent`
2. When running `EXPLAIN ANALYZE`

Stats will be missing when running a simple `EXPLAIN (TYPE DISTRIBUTED) ...`